File: blender-add-on/xrs/log.py
# SPDX-License-Identifier: Apache-2.0
# -*- coding: utf-8 -*-
""" Log related functions
"""

def debug(message):
  """ Print a color coded debug message to the console """
  print("\033[94m" + message + "\033[0m")

def error(message):
  """ Print a color coded error to the console """
  print("\033[91m" + "ERROR: " + message + "\033[0m")

def info(message):
  """ Print a color coded info message to the console """
  print("\033[92m" + message + "\033[0m")

def silly(message):
  """ Print a color coded silly message to the console """
  print("\033[95m" + message + "\033[0m")

def verbose(message):
  """ Print a color coded verbose message to the console """
  print("\033[96m" + message + "\033[0m")

def warn(message):
  """ Print a color coded warning to the console """
  print("\033[93m" + "Warning: " + message + "\033[0m")

File: website-services/app/log.ts
// SPDX-License-Identifier: Apache-2.0
const Winston = require('winston');

class Log {
  private warnOnlyFilter = Winston.format((info, opts) => {
    return info.level === 'warn' ? info : false;
  });
  private options = {
    error_file: {
      level: 'error',
      filename: './logs/error.log',
      format: Winston.format.combine(
        Winston.format.timestamp(),
        Winston.format.json(),
      ),
      handleExceptions: true,
      maxsize: 5242880,
      maxfiles: 5,
    },
    warning_file: {
      level: 'warn',
      filename: './logs/warnings.log',
      format: Winston.format.combine(
        this.warnOnlyFilter(),
        Winston.format.timestamp(),
        Winston.format.json(),
      ),
      handleExceptions: true,
      maxsize: 5242880,
      maxfiles: 5,
    },
    console: {
      level: 'silly',
      handleExceptions: true,
      format: Winston.format.combine(
        Winston.format.colorize(),
        Winston.format.simple(),
      ),
    },
  };
  private logger;
  constructor() {
    this.logger = Winston.createLogger({
      transports: [
        new Winston.transports.File(this.options.error_file),
        new Winston.transports.File(this.options.warning_file),
        new Winston.transports.Console(this.options.console),
      ],
      exitOnError: false,
    });
  }

  public error(message: String): void {
    this.logger.error(message);
  }

  public warn(message: String): void {
    this.logger.warn(message);
  }

  public info(message: String): void {
    this.logger.info(message);
  }

  public verbose(message: String): void {
    this.logger.verbose(message);
  }

  public debug(message: String): void {
    this.logger.debug(message);
  }

  public silly(message: String): void {
    this.logger.silly(message);
  }
}

module.exports = new Log();
export default new Log();
